<html>
    <head>
        <meta content="text/html; charset=UTF-8" http-equiv="content-type" />
        <link rel="stylesheet" href="css/computational-biology.css" />
        <link rel="stylesheet" href="css/page.css" />
    </head>

    <body class="c13">
        <p class="c16 title" id="h.2zjs6mbwkown">
            <span class="c15">Computational Biology</span>
        </p>
        <p class="c18 subtitle" id="h.rpxcjqfj0i86">
            <span class="c17">Joshua Gregory</span>
        </p>
        <h1 class="c7" id="h.e0gufr1llpyf">
            <span class="c12">1 - Introduction</span>
        </h1>
        <h2 class="c6" id="h.60skwgw1k2wa"><span class="c0">Data</span></h2>
        <ul class="c5 lst-kix_ysyb1hkcnow5-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Biology is big data</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">10&sup2; to 10&sup1;&sup2; numbers</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Lots of information stored in big databases</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.cr25r3yasaug">
            <span class="c0">Interactions</span>
        </h2>
        <ul class="c5 lst-kix_qjyg0wn0n63k-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Protein-protein common</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Protein-DNA for regulation</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Protein-environment how cell senses the world</span
                >
            </li>
            <li class="c2 li-bullet-0"><span class="c1">Ecosystems</span></li>
        </ul>
        <h2 class="c6" id="h.mqdmqwpntg1k">
            <span class="c0">Self-Organise</span>
        </h2>
        <ul class="c5 lst-kix_naso3p89xejj-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Protein complexes form machines</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Life can organise into cells</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Cells can organise into higher organisms</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Organisms form populations that act together</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Back to ecosystems</span>
            </li>
        </ul>
        <h2 class="c6" id="h.dnn3z0sbzqlk"><span class="c0">Dynamic</span></h2>
        <ul class="c5 lst-kix_jovu5xjwwzps-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Cells chemical machines</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Complex auto-regulation</span>
            </li>
            <li class="c2 li-bullet-0"><span class="c1">Cells divide</span></li>
            <li class="c2 li-bullet-0">
                <span class="c1">Can fuse sexually</span>
            </li>
            <li class="c2 li-bullet-0"><span class="c1">Life evolves</span></li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Almost all aspects of life governed by mechanisms</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.pgn4hl282s3f"><span class="c0">Comp Bio</span></h2>
        <ul class="c5 lst-kix_u5stges5iuz-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Comp bio is big data problem</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Tools in compsci essential to</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_u5stges5iuz-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Store the data</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Make inference about the data</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_u5stges5iuz-0">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Tools of abstraction apply to biology too</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.72nc4lyi9gea">
            <span class="c0">Evolution</span>
        </h2>
        <ul class="c5 lst-kix_ewtxwqac4ze-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Charles Darwin</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Natural Selection</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Still being understood</span>
            </li>
        </ul>
        <h2 class="c6" id="h.1n437sg8j6yi">
            <span class="c0">Tree of life</span>
        </h2>
        <ul class="c5 lst-kix_65p383xcps5n-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Life being related through tree like structure</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Molecularly, species share a huge amount in common</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Transfer knowledge between species</span>
            </li>
        </ul>
        <h2 class="c6" id="h.3y6r1pxu7t28">
            <span class="c0">Inheritance</span>
        </h2>
        <ul class="c5 lst-kix_z5rk2yjcv08u-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Not generally understood in Darwin&rsquo;s time</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Modern theory by Gregor Mendel</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Finally widely understood it occurred through exchange of
                    discrete units (genes)</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.qx5yb2pz039">
            <span class="c0">Mendel&rsquo;s Peas Experiment</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 356px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image41.png"
                    style="
                        width: 602px;
                        height: 356px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.uw6qhvkqnhm">
            <span class="c0">Inheritance</span>
        </h2>
        <ul class="c5 lst-kix_mlh9x6ly41c3-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Experiments explained if:</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_mlh9x6ly41c3-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Traits carried by discrete units (genes)</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1"
                    >Each individual carries two copies of a gene
                    (diploidy)</span
                >
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">We inherit one copy from each parent</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">One gene is dominant</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_mlh9x6ly41c3-0">
            <li class="c2 li-bullet-0">
                <span class="c1">These not always true</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Rigged experiment? Mechanism remained confused until
                    DNA</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.u8lxkhbyxdnr"><span class="c0">DNA</span></h2>
        <ul class="c5 lst-kix_sc2xpaeoql0g-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Deoxyribose Nucleic Acid</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Crick and Watson</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Discovery of double helix</span>
            </li>
        </ul>
        <h2 class="c6" id="h.ufeathg2vt0k">
            <span class="c0">Base Pairs</span>
        </h2>
        <ul class="c5 lst-kix_bi9msgn8tgvx-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Double helix held together by 4 types of bases: A, T, G,
                    C</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 270.11px;
                        height: 293.5px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image22.png"
                        style="
                            width: 270.11px;
                            height: 293.5px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
            <li class="c2 li-bullet-0">
                <span>Each base form </span><span class="c9">base pairs </span
                ><span class="c1">A-T or C-G</span>
            </li>
        </ul>
        <h2 class="c6" id="h.kwkfxevy51s">
            <span class="c0">DNA Replication</span>
        </h2>
        <ul class="c5 lst-kix_dn4hvxh0k0lm-0 start">
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 588.5px;
                        height: 371.48px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image49.png"
                        style="
                            width: 588.5px;
                            height: 371.48px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
        </ul>
        <h2 class="c6" id="h.d3hvl2wxiek9">
            <span class="c0">What does DNA do?</span>
        </h2>
        <ul class="c5 lst-kix_jz69xes7xjef-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Part codes for proteins</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_jz69xes7xjef-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1"
                    >Gets copied to messenger RNA through transcription</span
                >
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">mRNA then gets translated to proteins</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_jz69xes7xjef-0">
            <li class="c2 li-bullet-0">
                <span class="c1">A lot does not code for protein</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_jz69xes7xjef-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Can code for RNA molecules (ribozymes)</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Used for controlling the cell</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Can be junk - caused by replication</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Or we&rsquo;re not quite sure&hellip;</span>
            </li>
        </ul>
        <h2 class="c6" id="h.rvcegnshvvez">
            <span class="c0">RNA and DNA</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 484px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image31.png"
                    style="
                        width: 602px;
                        height: 484px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <ul class="c5 lst-kix_vxhsfypbjg6v-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">The base U is used in RNA instead of T</span>
            </li>
        </ul>
        <h2 class="c6" id="h.ju75lvv5m6g1">
            <span class="c0">Transcription by RNA Polymerase</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 365.33px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image51.png"
                    style="
                        width: 602px;
                        height: 365.33px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <ul class="c5 lst-kix_o2f0r1mvqj6n-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">DNA unwound, mRNA encoded, DNA rewound</span>
            </li>
        </ul>
        <h2 class="c6" id="h.wbp0kcgq8jc8">
            <span class="c0">Exons and Introns</span>
        </h2>
        <ul class="c5 lst-kix_utjn4r4v0die-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Exons are regions in DNA that will encode the RNA</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Introns are the noncoding regions</span>
            </li>
        </ul>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 363.5px;
                    height: 231.87px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image24.png"
                    style="
                        width: 363.5px;
                        height: 231.87px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.syqn5ze2vlig">
            <span class="c0">mRNA to Protein Translation</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 493.5px;
                    height: 404.15px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image11.png"
                    style="
                        width: 493.5px;
                        height: 404.15px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <ul class="c5 lst-kix_2lxm39fubhol-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Ribosome moves across the mRNA strand</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >The groups of 3 nucleotides (codons) act as the code for
                    amino acid molecules</span
                >
            </li>
        </ul>
        <ul class="c5 lst-kix_2lxm39fubhol-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1"
                    >Can have multiple codes for the same acid</span
                >
            </li>
        </ul>
        <ul class="c5 lst-kix_2lxm39fubhol-0">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >The amino acids attached together in the right order</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Then the strand of acids form the protein</span
                >
            </li>
        </ul>
        <p class="c3 c4"><span class="c1"></span></p>
        <ul class="c5 lst-kix_ckcga7slzd9k-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >DNA nucleotides -&gt; mRNA nucleotides -&gt; amino acids
                    -&gt; protein</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.iihqsv6lyydi">
            <span class="c0">RNA Triplet to Amino Acids</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 636px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image18.png"
                    style="
                        width: 602px;
                        height: 636px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <ul class="c5 lst-kix_3s3t2puaricr-0 start">
            <li class="c2 li-bullet-0">
                <span class="c10">Transcription</span
                ><span class="c1">: DNA -&gt; RNA</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c10">Translation</span
                ><span class="c1">: RNA -&gt; Protein</span>
            </li>
        </ul>
        <p class="c3 c4"><span class="c1"></span></p>
        <p class="c3"><span class="c1">To help remember(?):</span></p>
        <ul class="c5 lst-kix_glvpm2h38enf-0 start">
            <li class="c2 li-bullet-0">
                <span>Transc</span><span class="c9">R</span
                ><span>iption - creating </span><span class="c9">R</span
                ><span class="c1">NA, transcribing the DNA to RNA</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Translation - translating the language of RNA to the
                    language of proteins</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.k9gmrjp4q4ru">
            <span class="c0">What do proteins do?</span>
        </h2>
        <ul class="c5 lst-kix_zd5xbr2w9m3f-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Main building block of cells</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Form skeleton (lipid layer of membranes)</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Act as chemical machines (enzymes)</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Act as mechanical &nbsp;machines</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Large part of control mechanism</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Not everything in cell is protein</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_zd5xbr2w9m3f-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1">There is DNA and RNA</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1"
                    >Rna can fold up like proteins and act as catalysts
                    (ribosymes)</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.pa1xzb5k6m1v">
            <span class="c0">Computational challenges</span>
        </h2>
        <ul class="c5 lst-kix_dl3dqibaezqk-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >A lot of challenges when trying to understand genome</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Need to find coding regions (Exons)</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Need to reconstruct DNA sequence from sequencing data</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Need to compare different sequences</span>
            </li>
        </ul>
        <h1 class="c7" id="h.8git9jer2ye0">
            <span>2 - </span><span class="c12">Cell Regulation</span>
        </h1>
        <h2 class="c6" id="h.fkbjnkvnlgpy"><span class="c0">Cells</span></h2>
        <ul class="c5 lst-kix_ey4wyq6fc4rz-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Complicated machines</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Still on-going understanding</span>
            </li>
        </ul>
        <h2 class="c6" id="h.oezclba5hs0j">
            <span class="c0">Prokaryotes and Eukaryotes Cell Types</span>
        </h2>
        <ul class="c5 lst-kix_57b7vtlboypb-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Share cell membrane, DNA code, ribosome (machine for
                    producing proteins)</span
                >
            </li>
        </ul>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 268px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image19.png"
                    style="
                        width: 602px;
                        height: 268px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <ul class="c5 lst-kix_57b7vtlboypb-0">
            <li class="c2 li-bullet-0"><span class="c1">Eukaryotes</span></li>
        </ul>
        <ul class="c5 lst-kix_57b7vtlboypb-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Has a true nucleus</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1"
                    >Multiple linear chromosomes in the nucleus</span
                >
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Larger and more complex</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Formed by fusing different cells</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Can be multicellular</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">e.g. plant and animal cells</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_57b7vtlboypb-0">
            <li class="c2 li-bullet-0"><span class="c1">Prokaryotes</span></li>
        </ul>
        <ul class="c5 lst-kix_57b7vtlboypb-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Do not have a nucleus</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Single DNA molecule in nucleoid</span>
            </li>
            <li class="c3 c11 li-bullet-0"><span class="c1">Smaller</span></li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Usually unicellular</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">e.g. bacteria cell</span>
            </li>
        </ul>
        <p class="c3 c4"><span class="c1"></span></p>
        <h2 class="c6" id="h.2mbewkmwpbtx">
            <span class="c0">Understanding proteins</span>
        </h2>
        <ul class="c5 lst-kix_cw1ciuid3qc8-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Hard to understand and sequence directly</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Deduce their sequence from DNA or mRNA</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Measuring protein concentration difficult:</span
                >
            </li>
        </ul>
        <ul class="c5 lst-kix_cw1ciuid3qc8-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Field of proteomics</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Uses mass spectrometers to count</span>
            </li>
        </ul>
        <h2 class="c6" id="h.5ou04imueho6">
            <span class="c0">Protein Folding</span>
        </h2>
        <ul class="c5 lst-kix_z9kghg5m2pg-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Understand their 3D shape</span>
            </li>
            <li class="c2 li-bullet-0"><span class="c1">Difficult</span></li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Crystallise them and use X-ray diffraction</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Predicting structure is non-local problem</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.k0b7ba6j2svo">
            <span class="c0">(Primary Structure)</span>
        </h2>
        <ul class="c5 lst-kix_5myrrsg21k1c-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">The sequence/order of amino acids</span>
            </li>
        </ul>
        <h2 class="c6" id="h.lo00nr9k00v8">
            <span class="c0">Secondary Structure</span>
        </h2>
        <ul class="c5 lst-kix_pasbsn9v3trw-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Main components make up proteins:</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_pasbsn9v3trw-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Alpha helices</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Beta sheets</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Various folds and connecting parts</span>
            </li>
        </ul>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 249.94px;
                    height: 297.5px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image26.png"
                    style="
                        width: 249.94px;
                        height: 297.5px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.xf97vi1l4z55"><span class="c0">Folding</span></h2>
        <ul class="c5 lst-kix_qrcipjrpc90m-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Protein folding is hard because typically amino acids apart
                    will interact through hydrogen bonding</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">RNA also folds (more flexible than DNA)</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">RNA folding easier to solve</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Look for matching sub-sequences</span>
            </li>
        </ul>
        <h2 class="c6" id="h.2hpujvz80o9l">
            <span class="c0">RNA Folding</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 388.5px;
                    height: 292.34px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image42.png"
                    style="
                        width: 388.5px;
                        height: 292.34px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.2u73d2rxa8hg">
            <span class="c0">The First Law of Protein Sequence</span>
        </h2>
        <p class="c3">
            <span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span
            ><span class="c14 c9"
                >Similar structures perform similar functions</span
            >
        </p>
        <p class="c3 c4"><span class="c14 c9"></span></p>
        <ul class="c5 lst-kix_267msnhfipc7-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Learn about function of protein by comparing it with
                    others</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Evolution means many proteins are common to huge number of
                    organisms</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Converse not necessarily true: dissimilar structures can
                    have similar function</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.exx4hr9vckp5">
            <span class="c0">Sequence Alignment</span>
        </h2>
        <ul class="c5 lst-kix_de1dcpsu7hu7-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Gene sequences aligned and similarities compared</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">e.g. mouse gene to shark gene</span>
            </li>
        </ul>
        <h2 class="c6" id="h.jolfz5kcfntw">
            <span class="c0">Finding Aligned Sequences</span>
        </h2>
        <ul class="c5 lst-kix_drgi6c8ea69u-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Compare new sequences with ones already found</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Compare with large database</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >They are usually annotated indicating what someone thinks a
                    protein does in some organism</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.81zsi0een558">
            <span class="c0">Cell Regulation</span>
        </h2>
        <ul class="c5 lst-kix_y9l22fl11108-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Cells must respond to environment, maintain their
                    environment, divide</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Need control mechanism, proteins and DNA help with
                    this</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.ipay69vfb0r8">
            <span class="c0">Promoter Region</span>
        </h2>
        <ul class="c5 lst-kix_jhiu5nrt7p3-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Basically this is a region in the DNA which helps regulate
                    the levels of proteins in the cell</span
                >
            </li>
        </ul>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 348px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image23.png"
                    style="
                        width: 602px;
                        height: 348px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.l97g46ifnuuc">
            <span class="c0">Regulation by Signal</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 284px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image30.png"
                    style="
                        width: 602px;
                        height: 284px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <p class="c3 c4"><span class="c1"></span></p>
        <h2 class="c6" id="h.acyrb4ee33nj">
            <span class="c0">Regulation in Promoter Region</span>
        </h2>
        <ul class="c5 lst-kix_x0w11mg24zvl-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Many different proteins can bind to the promoter
                    region</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >They can either excite the expression of the protein (by
                    increasing the probability of RNA-polymerase binding)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Or inhibit the expression of the protein</span>
            </li>
        </ul>
        <h2 class="c6" id="h.m13k23ge5lqe">
            <span class="c0">Binding Sites</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 354.67px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image32.png"
                    style="
                        width: 602px;
                        height: 354.67px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <p class="c3"><span class="c1">?</span></p>
        <h2 class="c6" id="h.h8e8o62pljsy"><span class="c0">TATA Box</span></h2>
        <ul class="c5 lst-kix_nrxid2h2i5ks-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Close to the coding regions is a region where are more A-T
                    pairs than usual</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Because A-T pairs have only 2 hydrogen bonds, they are
                    easier to break</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >This helps RNA-polymerase split open the double helix</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.kazcum2o1ud5">
            <span class="c0">Gene Regulation Networks</span>
        </h2>
        <ul class="c5 lst-kix_b6sj7yrdihl-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Proteins can regulate their own expression and many other
                    proteins</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span>Protein </span><span class="c10">A</span
                ><span>&nbsp;can regulate protein </span
                ><span class="c10">B</span
                ><span>&nbsp;which in turn regulates protein </span
                ><span class="c10">A</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >In general regulation forms complex circuit</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.jb2qgegxakr2"><span class="c0">Motifs</span></h2>
        <ul class="c5 lst-kix_wtu2283hd0qr-0 start">
            <li class="c2 li-bullet-0">
                <span
                    >To understand what&rsquo;s going on we can look for
                    patterns or </span
                ><span class="c9">motifs</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Spot common motifs by askings what patterns happen more
                    than by chance</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Synthetic biology builds biological circuits from these
                    motifs</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.isto54ukvx2m"><span class="c0">Dynamics</span></h2>
        <ul class="c5 lst-kix_hsxziq72bak5-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Gene regulation networks are dynamic</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">To understand: simulate the network</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Requires knowledge of rate at which molecules are being
                    produced</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.5o2ux3eewkyg">
            <span class="c0">Multicellular</span>
        </h2>
        <ul class="c5 lst-kix_ifr44k9z9lim-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Many Eukaryotes are multicellular</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Many cases the cells differentiate</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">How do they know what role they play?</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >How do organisms develop so cells do the right thing?</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">You&rsquo;ll find out later</span>
            </li>
        </ul>
        <h1 class="c7" id="h.d0l5etpkc1w9">
            <span class="c12">3 - Sequencing</span>
        </h1>
        <h2 class="c6" id="h.bn2ozpxr9q7i">
            <span class="c0">What is it?</span>
        </h2>
        <ul class="c5 lst-kix_kk8r7kvbd3bv-0 start">
            <li class="c2 li-bullet-0">
                <span
                    >DNA sequencing is the process of determining the nucleic
                    acid sequence &ndash; the order of nucleotides in DNA.</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.emyl25di2tlu">
            <span class="c0">Sequencing Revolution</span>
        </h2>
        <ul class="c5 lst-kix_yyh9f5ltv5yj-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Advances in DNA sequencing</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Reading DNA took a while to take off since structure
                    discovery in 1950s</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Francis Crick had speculated on how DNA might code for
                    proteins</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >By removing some number of bases from DNA he was able to
                    show that DNA has a three letter code</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.1qcgxm4kn8iv">
            <span class="c0">Frederick Sanger</span>
        </h2>
        <ul class="c5 lst-kix_s7mhqd5maa2-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">British biochemist</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >1995 discovered protein sequence for Insulin</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Showed proteins actually chains</span>
            </li>
        </ul>
        <h2 class="c6" id="h.qvutasuzie80">
            <span class="c0">Sequencing DNA</span>
        </h2>
        <ul class="c5 lst-kix_sqtapf55jivx-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Chemically difficult as bases not distinctive</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">1965 when first strands of RNA read</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Copying DNA and breaking into smaller sequences was
                    effort</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Various techniques developed in 1970s</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">1977 &ldquo;Sanger Sequencing&rdquo;</span>
            </li>
        </ul>
        <h2 class="c6" id="h.7e5yyc52zech">
            <span class="c0">Sanger Sequencing</span>
        </h2>
        <ul class="c5 lst-kix_nfqjnmpxh8s-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Use biological mechanisms for copying DNA</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Would add Dideoxynucleoties (ddNTP) to a DNA chain</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.zgmylnp1gm9k">
            <span class="c0">Modified ddNTP</span>
        </h2>
        <ul class="c5 lst-kix_pvo9ln7htu24-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Developed with a missing 3&rsquo; hydroxyl group which
                    prevented chain from developing any further</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >These marked (with radioactive phosphorus or later with
                    fluorescent marker)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >DNA replication carried out with normal ddNTPs and a weak
                    concentration of these modified ddNTPs</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span
                    >Initially done in 4 containers, one for each base (in
                    container </span
                ><span>A only base </span><span class="c10">A</span
                ><span class="c1">&nbsp;ddNTP were modified)</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Later different fluorescence were used so everything could
                    be carried out in the same vessel</span
                >
            </li>
        </ul>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 265.33px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image5.png"
                    style="
                        width: 602px;
                        height: 265.33px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <ul class="c5 lst-kix_a1rz4stz46uq-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">* means modified ddNTP</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >When DNA replicating, the chain would stop at these *
                    points</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">You could identify these points</span>
            </li>
        </ul>
        <h2 class="c6" id="h.iged5n3ivo9q">
            <span class="c0">Fractionation</span>
        </h2>
        <ul class="c5 lst-kix_v32bz2hlms48-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Each solution added to a layer of gel</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Using capillary electrophoresis different length DNA
                    fragments are sorted</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >From the distance travelled and the colour of the last
                    nucleotide base we can read off the sequence</span
                >
            </li>
        </ul>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 199.85px;
                    height: 414.5px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image25.png"
                    style="
                        width: 199.85px;
                        height: 414.5px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.mvwcyjc7gz8k">
            <span class="c0">Shotgun Sequencing</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 350.67px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image39.png"
                    style="
                        width: 602px;
                        height: 350.67px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <ul class="c5 lst-kix_bfladi6dnaoy-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Saves careful preparation</span>
            </li>
        </ul>
        <h2 class="c6" id="h.q33r31dx6kcl">
            <span class="c0">Reconstruction</span>
        </h2>
        <ul class="c5 lst-kix_p7p170rkje3u-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Looks easy but genome very repetitive</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Sanger only allows you to sequence &ldquo;reads&rdquo; of
                    around 1000bp (base pairs)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Not clear what long range ordering is</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Larger genomes, &ldquo;scaffold&rdquo; built where position
                    of some genes knew</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Sequence both dies of large fragment so you know
                    approximate relationship</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.pl4rp8380jhm">
            <span class="c0">Human Genome Project</span>
        </h2>
        <ul class="c5 lst-kix_g74l9978mf7p-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Sanger Sequencing helped human genome project</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Costs were predicted high</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Still don&#39;t know enough to provide all benefits</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.1dg1quicvaj6">
            <span class="c0">Improving Sanger</span>
        </h2>
        <ul class="c5 lst-kix_r4g08zqfriz5-0 start">
            <li class="c2 li-bullet-0"><span class="c1">Costs a lot</span></li>
            <li class="c2 li-bullet-0">
                <span class="c1">Improve with massive parallelisation</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Many techniques used</span>
            </li>
        </ul>
        <h2 class="c6" id="h.o3zbk05e3ht6">
            <span class="c0">Sequencing Techniques</span>
        </h2>
        <ul class="c5 lst-kix_ymvvgwm9eyt-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Sanger: sequencing-by-synthesis</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_ymvvgwm9eyt-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1"
                    >Using DNA-polymerase a strand of DNA is replicated</span
                >
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1"
                    >It turns out this technique ultimately triumphant</span
                >
            </li>
        </ul>
        <ul class="c5 lst-kix_ymvvgwm9eyt-0">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Another approach is sequence-by-ligation (or
                    hybridisation)</span
                >
            </li>
        </ul>
        <ul class="c5 lst-kix_ymvvgwm9eyt-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1"
                    >Create lots of short sections of DNA (oligonucleotides)
                    that will stick to an unbound stretch of DNA (or RNA)</span
                >
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Use in microarrays</span>
            </li>
        </ul>
        <h2 class="c6" id="h.h3fkwryh5wbu">
            <span class="c0">Synthesis vs Ligation</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 517.33px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image6.png"
                    style="
                        width: 602px;
                        height: 517.33px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.ydqq5yqctqem">
            <span class="c0">454 Sequencer</span>
        </h2>
        <ul class="c5 lst-kix_mlt1yb62dgic-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >First high throughput sequencer was Roche 454 machine</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Amplified fragments of DNA attached to a microbead</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">These are contained in mirco-wells</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Sequencing-by-synthesis is carried out</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Added one type of base at a time with fluorescent
                    label</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Can have difficulty distinguishing regions with repeated
                    base</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.lkqz24pg4ebp">
            <span class="c0">Water in Oil Emulsion PCR</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 591.5px;
                    height: 443.93px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image48.png"
                    style="
                        width: 591.5px;
                        height: 443.93px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.7z7l6j7c0jho"><span class="c0">Illumina</span></h2>
        <ul class="c5 lst-kix_z3x1mm7kxop1-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >The dominant next generation machines are produced by
                    Illumina (formally Solexa)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Uses technique called bridge amplification to create
                    clusters of identical strands on a 2-D surface
                    (substrate)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Can sequence 100s of million clusters at once</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >It then uses sequencing-by-synthesis to read the
                    sequence</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Also read the other end of the sequence by flipping
                </span>
            </li>
        </ul>
        <h2 class="c6" id="h.pkps6jcbgvo2">
            <span class="c0">Bridge Amplification</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 357.33px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image21.png"
                    style="
                        width: 602px;
                        height: 357.33px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.t954pqdpmv03">
            <span class="c0">Identifying Sequence</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 429.33px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image16.png"
                    style="
                        width: 602px;
                        height: 429.33px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <p class="c3"><span class="c1">? low res :(</span></p>
        <h2 class="c6" id="h.8o6pfm9uh92b">
            <span class="c0">Cost Per Human Genome</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 448px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image35.png"
                    style="
                        width: 602px;
                        height: 448px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <ul class="c5 lst-kix_swevwg42f21n-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">2007, 2015 jump</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Surpassed Moore&rsquo;s Law</span>
            </li>
        </ul>
        <h2 class="c6" id="h.4edknl1i6z90">
            <span class="c0">100,000 Genome Project</span>
        </h2>
        <ul class="c5 lst-kix_psbw075iuinn-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Currently NHS and Sanger centre sequencing around 70,000
                    human genome to identify rare genetic diseases</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Can now sequence a human genome within a day</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >As already sequenced human genome, no longer have
                    difficulty reconstructing sequence</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.9t86cqp4s6hq">
            <span class="c0">Single Molecule DNA Sequencing</span>
        </h2>
        <ul class="c5 lst-kix_9pv0cxxw7rf5-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Replicating DNA can bring errors and biases</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Most high-throughput techniques only read short sections
                    (around 100bp), reconstruction difficult</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Current research to create machine that can read long
                    single DNA strand</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.60urmt66mxfw">
            <span class="c0">Zero Mode Waveguide</span>
        </h2>
        <ul class="c5 lst-kix_yuplr44rmqe0-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Uses same fluorescence dNTPs as used in high-throughput
                    processing</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Use micro-well smaller than the wavelength of light</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Able to detect very small signal</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Not high-throughput</span>
            </li>
        </ul>
        <h2 class="c6" id="h.530u9v1w0c0y">
            <span class="c0">Nanopore Technology</span>
        </h2>
        <ul class="c5 lst-kix_r7fwi48ah90j-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Force DNA through tiny pore and read out the sequence</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Hand-held machines</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Currently accuracy not quite good enough</span>
            </li>
        </ul>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 583.96px;
                    height: 392.5px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image43.png"
                    style="
                        width: 583.96px;
                        height: 392.5px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 455.5px;
                    height: 372.45px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image13.png"
                    style="
                        width: 455.5px;
                        height: 372.45px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h1 class="c7" id="h.f1eps4brwrdn">
            <span class="c12">4 - Sequence Re-assembly</span>
        </h1>
        <h2 class="c6" id="h.ok33rnr82zlw">
            <span class="c0">DNA Sequencing</span>
        </h2>
        <ul class="c5 lst-kix_cav74gfbzkgr-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >In shotgun genome sequencing, full genome broken into small
                    pieces</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Pieces then read by a sequencing machine</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Reads short sections (100-1000) bases</span>
            </li>
            <li class="c2 li-bullet-0">
                <span>The </span><span class="c9">reads </span
                ><span class="c1"
                    >are then assembled to construct the full genome</span
                >
            </li>
        </ul>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 240px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image45.png"
                    style="
                        width: 602px;
                        height: 240px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <p class="c3">
            <span class="c1"
                >Overlapping the pieces to reconstruct. Need algorithms to do
                this</span
            >
        </p>
        <h2 class="c6" id="h.6t18bp2muuwy">
            <span class="c0">Reads and Contigs</span>
        </h2>
        <ul class="c5 lst-kix_d6tfzsi4a98e-0 start">
            <li class="c2 li-bullet-0">
                <span>We are given set of </span><span class="c9">reads </span
                ><span class="c1">(DNA segments)</span>
            </li>
            <li class="c2 li-bullet-0">
                <span>We need to reconstruct them into </span
                ><span class="c9">contigs </span
                ><span class="c1"
                    >(set of overlapping DNA segments that form a consensus
                    region of DNA)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Problem as (0.1-2%) errors in reads</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Shotgun important in speeding up human genome project</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">What data structure to use?</span>
            </li>
        </ul>
        <h2 class="c6" id="h.dk8zwzpi3vae">
            <span class="c0">Digital Trees or Tries</span>
        </h2>
        <ul class="c5 lst-kix_p9dof6tjj2tx-0 start">
            <li class="c2 li-bullet-0">
                <span>A </span><span class="c9">Trie</span
                ><span>&nbsp;(pronounced: &ldquo;try&rdquo;) or </span
                ><span class="c9">digital tree</span
                ><span class="c1"
                    >&nbsp;is a multiway tree data structure often used for
                    storing large sets of words</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >They are trees with a possible branch for every letter of
                    an alphabet</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span>Their names comes from re</span
                ><span class="c9">trie</span><span class="c1">val</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Tries usually compactify the edges in the tree</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >All words end up with a special letter
                    &ldquo;$&rdquo;</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.unwnejwn73id">
            <span class="c0">Trie example</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 170.67px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image40.png"
                    style="
                        width: 602px;
                        height: 170.67px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.vpeudzuccd84">
            <span class="c0">General Use of Tries</span>
        </h2>
        <ul class="c5 lst-kix_nzdrgpgkqish-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Implementation of sets</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Quick insertion, deletion and find operations</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Typically considerably quicker than binary trees and hash
                    tables</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Good for spell checkers, completion algorithms,
                    longest-prefix matching, hyphenation</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Each search finds longest match between the words in the
                    set and the query</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.7oxdw1qu9684">
            <span class="c0">Tries for Sequence Analysis</span>
        </h2>
        <ul class="c5 lst-kix_am9sekbobrmh-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Tries can be used to store DNA or proteins</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >For reconstruction of DNA tries are very useful for finding
                    the next fitting section</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Useful when there are errors, as by searching small
                    sub-tree you can find good match</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >For DNA you have a four-way (5 way?) tree, which means your
                    tree is deep but it doesn&rsquo;t waste so much memory</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.bodtc04lsexa">
            <span class="c0">Example table</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 369.33px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image29.png"
                    style="
                        width: 602px;
                        height: 369.33px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.3813ugg2moa">
            <span class="c0">Disadvantages of Tries</span>
        </h2>
        <ul class="c5 lst-kix_7yobbprdnm0r-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Table-based typically waste large amounts of memory</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Often table-based used for first few layers, while lower
                    levels less memory intensive structure</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >These days memory less of a problem, so table used</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Many implementations of tries, each suited to particular
                    task</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >For DNA, tries not too wasteful (maybe though for
                    proteins)</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.ukpjptfritt">
            <span class="c0">Suffix Tree</span>
        </h2>
        <ul class="c5 lst-kix_ya682lvvd12o-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Suffix tree is a trie of all suffixes of a string</span
                >
            </li>
            <li class="c2 li-bullet-0"><span class="c1">E.g. banana</span></li>
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 602px;
                        height: 226.67px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image8.png"
                        style="
                            width: 602px;
                            height: 226.67px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
        </ul>
        <h2 class="c6" id="h.nv5ndtvzkrr4">
            <span class="c0">Importance of Suffix Tree</span>
        </h2>
        <ul class="c5 lst-kix_xt3wkmi6q11o-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Fast algorithms to compute suffix trees (and the related
                    suffix array) are relatively new data structures</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Important for string-based algorithms</span>
            </li>
            <li class="c2 li-bullet-0">
                <span
                    >Classic application is in finding a match for a query
                    string, </span
                ><span class="c10">Q</span><span>, in a text, </span
                ><span class="c10">T</span>
            </li>
        </ul>
        <h2 class="c6" id="h.l6g1o0ylc53">
            <span class="c0">String Matching</span>
        </h2>
        <ul class="c5 lst-kix_5zfzhtm95vq1-0 start">
            <li class="c2 li-bullet-0">
                <span>To find a match of a query string, </span
                ><span class="c10">Q = an</span><span>, in a text, </span
                ><span class="c10">T = banana</span
                ><span
                    >, we can first construct the suffix tree of the string </span
                ><span class="c10">T</span><span class="c1">&nbsp;</span>
            </li>
            <li class="c2 li-bullet-0">
                <span>We then simply look up the query, </span
                ><span class="c10">Q</span
                ><span class="c1">, using the trie</span>
            </li>
        </ul>
        <p class="c3"><span class="c1">(Image above)</span></p>
        <h2 class="c6" id="h.2uafpg5206f4">
            <span class="c0">Complexity of Suffix Tree</span>
        </h2>
        <ul class="c5 lst-kix_82aqlr7v1w19-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Using a regular trie for a suffix tree would typically use
                    far too much memory to be useful</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span
                    >However, by using pointers to the original text it is
                    possible to build a suffix tree using O(n) memory where </span
                ><span class="c10">n</span
                ><span class="c1">&nbsp;is the length of the text</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Furthermore there is a linear time (O(n)) algorithm to
                    construct the trie</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >The algorithm is not however trivial to understand</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.e5bhngxp1lar">
            <span class="c0">Uses of Suffix Trees</span>
        </h2>
        <ul class="c5 lst-kix_1nbswr2j2m1x-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Suffix trees are efficient whenever it is likely that you
                    will do multiple searches</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Exact word matching is in itself is a very important
                    application</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Suffix trees in combination with dynamic programming can be
                    used to do inexact matching (finding the match with the
                    smallest edit distance)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Suffix trees get used in bioinformatics, advanced machine
                    learning algorithms,...</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">(Prefix trees also exist as well lol)</span>
            </li>
        </ul>
        <h2 class="c6" id="h.sat947werzne">
            <span class="c0">New Generation Sequencers</span>
        </h2>
        <ul class="c5 lst-kix_j6jdjoe3iqtb-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Estimated cost in 2005 of genome sequencing was
                    $10,000,000</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >To reduce cost, produce new sequencing machines</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Read shorter sections of DNA (20-100bp)</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Can these be re-assembled?</span>
            </li>
        </ul>
        <h2 class="c6" id="h.qreq6dba9iw6"><span class="c0">Repeats</span></h2>
        <ul class="c5 lst-kix_3wufi1mgzok-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Difficulty of assembly is caused by repeates</span
                >
            </li>
        </ul>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 26.67px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image4.png"
                    style="
                        width: 602px;
                        height: 26.67px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <ul class="c5 lst-kix_hohogkydakgk-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >How many repeats are there in human genome? (3.2 billions
                    base pairs)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Can read DNA in either directions, meaning effectively 6.4
                    billion bases</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Important question for developing new technologies</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.bbfc282ed443">
            <span class="c0">Repeats in Human Genome</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 423.73px;
                    height: 400.5px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image9.png"
                    style="
                        width: 423.73px;
                        height: 400.5px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.nv6jngmytf73">
            <span class="c0">Repeats Structure</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 493.28px;
                    height: 397.5px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image38.png"
                    style="
                        width: 493.28px;
                        height: 397.5px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.e9x8zc9isaxb">
            <span class="c0">Computing Repeats</span>
        </h2>
        <ul class="c5 lst-kix_nrhxq261wpk9-0 start">
            <li class="c2 li-bullet-0">
                <span>A naive program would take </span
                ><span class="c10">n</span><span>&sup2; operations where </span
                ><span class="c10">n</span
                ><span class="c1">&nbsp;= 6.4*10&#8313;</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">This is too long</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Couldn&rsquo;t answer this question a few years ago, not
                    because of computer power, but as algorithms not been
                    developed</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >However using suffix arrays counting repeats was made
                    easy</span
                >
            </li>
        </ul>
        <h1 class="c7" id="h.o9ny38rhyg4f">
            <span class="c12">5 - Sequence Matching</span>
        </h1>
        <h2 class="c6" id="h.g22t1pna88li">
            <span class="c0">Alignment and Evolution</span>
        </h2>
        <ul class="c5 lst-kix_6hcazc1dsg7g-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >We want to transfer our knowledge about the function of a
                    protein through knowledge of similar proteins in different
                    organisms</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Most proteins found in organism are close to those of near
                    relatives and sometimes not so near relatives</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.gw5umvpdsjex">
            <span class="c0">Practical Alignment</span>
        </h2>
        <ul class="c5 lst-kix_6vlrlc23kesr-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Alignment with known genomes is now the first task after
                    sequencing</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Done with protein and DNA</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Mutation rate sufficiently high that expect no alignment
                    between most species except if there is selective pressure
                    to conserve the structure</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >For DNA coding proteins there are a number of
                    &ldquo;synonymous mutations&rdquo; that leave the protein
                    unchanged - DNA is less conserved than proteins</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >In proteins many amino acids behave similarly so there is
                    little selection pressure against mutations</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.ogvmpmw75t7h">
            <span class="c0">Synonymous Mutations</span>
        </h2>
        <ul class="c5 lst-kix_7t3d8ft0jloh-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Some amino acids are rarer than others</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >But some codon sequences are slightly more popular than
                    their synonymouse partners</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">They may be more robust to mutation</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >They may be slight biases due to the translation,
                    etc.</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.tb1r0gepgyjy">
            <span class="c0">Amino Acid Similarity</span>
        </h2>
        <ul class="c5 lst-kix_9qospid47dwt-0 start">
            <li class="c2 li-bullet-0">
                <span
                    >Most important distinguishing property is whether they are </span
                ><span class="c9">hydrophobic </span><span>(oily) or </span
                ><span class="c9">hydrophilic</span
                ><span class="c1">&nbsp;(charged)</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Other properties such as hydrogen bonds, etc.</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Many observations can compute a table giving log-odds of
                    one amino acids being substituted by another</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Two commonly used substitution tables (matrices) are PAM
                    and BLOSUM</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Likely substitutions are positive, unlikely ones are
                    negative</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.49iwo5r8q4iu"><span class="c0">Blosum50</span></h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 405.33px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image12.png"
                    style="
                        width: 602px;
                        height: 405.33px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <ul class="c5 lst-kix_ja8obq8ij69c-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Each letter represents an amino acid</span>
            </li>
        </ul>
        <h2 class="c6" id="h.wl70npld80hs"><span class="c0">Gaps</span></h2>
        <ul class="c5 lst-kix_do0r4uib04g-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >It is not infrequent that we find good matches where there
                    has been an insertion or deletion of some part of the
                    string</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >We start by considering a linear cost where the cost of a
                    gap is proportional to its length</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >In biology, a linear cost model is a poor description of
                    reality</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Where there is an insertion, then it is often quite long
                    (this is a consequence of the copy mechanism in cells)</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.lgeb293ygmwq">
            <span class="c0">Inexact Sequence Matching</span>
        </h2>
        <ul class="c5 lst-kix_6goylr3qanow-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Given 2 sequences we want to find the best match between
                    them</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span>We can measure the distance in terms of an </span
                ><span class="c9">edit distance</span><span>&nbsp;or </span
                ><span class="c9">Levenshtein </span
                ><span class="c1"
                    >distance which measure the minimum number of moves to go
                    from one sequence to another</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >There are two cases that are commonly considered</span
                >
            </li>
        </ul>
        <ul class="c5 lst-kix_6goylr3qanow-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1"
                    >Global matching: where we match the two complete
                    strings</span
                >
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1"
                    >Local matching: we find the sub-string with the best
                    match</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.30bm0e6cs0mo">
            <span class="c0">Algorithms for Matching</span>
        </h2>
        <ul class="c5 lst-kix_ijei8ujuw48d-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Need efficient algorithms for matching</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Number of possible inexact matches is exponential in the
                    size of the string so we can&rsquo;t exhaustively try all
                    matches</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Fortunately dynamical programming provides a relative
                    efficient algorithm</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Dynamic programming is very important to a huge number of
                    sequence-based algorithms</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.dgrra3j6dpy">
            <span class="c0">Dynamic Programming</span>
        </h2>
        <ul class="c5 lst-kix_lmcv4oil4mk2-0 start">
            <li class="c2 li-bullet-0">
                <span
                    >Powerful strategies for solving optimisation problems is </span
                ><span class="c9 c14">dynamic programming</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_lmcv4oil4mk2-1 start">
            <li class="c3 c11 li-bullet-0">
                <span class="c1">Build set of optimal partial solutions</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1"
                    >Increase size of the partial solutions until have full
                    solution</span
                >
            </li>
            <li class="c3 c11 li-bullet-0">
                <span class="c1"
                    >Each step uses the set of optimal partial solutions found
                    in the previous step</span
                >
            </li>
        </ul>
        <ul class="c5 lst-kix_lmcv4oil4mk2-0">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >(name bad as doesn&#39;t actually have much to do with
                    programming)</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.ti7a1gv12sb0">
            <span class="c0">A Toy Problem</span>
        </h2>
        <ul class="c5 lst-kix_8oiirce68h7p-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Find minimum cost path from (0,0) to (8,0) though
                    this:</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 258px;
                        height: 254px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image7.png"
                        style="
                            width: 258px;
                            height: 254px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
            <li class="c2 li-bullet-0"><span class="c1">Costs in red</span></li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Cost of path is sum of weights on each link</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.bmvo30trguf2">
            <span class="c0">Brute Force</span>
        </h2>
        <ul class="c5 lst-kix_rq3zm5xdocu2-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">Try every path</span>
            </li>
            <li class="c2 li-bullet-0">
                <span>Problem with </span><span class="c10">n</span
                ><span>&nbsp;steps we require </span><span class="c10">n</span
                ><span>/2 to be diagonally up and </span
                ><span class="c10">n</span
                ><span class="c1">/2 to be diagonally down</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">The total number of paths is:</span>
            </li>
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 230px;
                        height: 88px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image2.png"
                        style="
                            width: 230px;
                            height: 88px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
            <li class="c2 li-bullet-0">
                <span>With </span><span class="c10">n </span
                ><span class="c1">= 8 there are 70 paths</span>
            </li>
            <li class="c2 li-bullet-0">
                <span>With </span><span class="c10">n</span
                ><span class="c1"
                    >&nbsp;= 100 there are 1.01*10&sup2;&#8313; paths</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.vs3twzh7dq8t">
            <span class="c0">Building a solution</span>
        </h2>
        <ul class="c5 lst-kix_krm5w0s0g33u-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Solve problem efficiently using dynamic programming by
                    considering optimal paths of shorter length</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span>Let c</span><span>(i,j)</span
                ><span class="c1"
                    >&nbsp;denote cost of optimal path to node (i,j)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >We denote the weights between two points on the lattice
                    by:</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 462px;
                        height: 198px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image3.png"
                        style="
                            width: 462px;
                            height: 198px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">We define initial cost c(0,0) = 0</span>
            </li>
        </ul>
        <h2 class="c6" id="h.4pnkptr70vs3">
            <span class="c0">Forward Algorithm</span>
        </h2>
        <ul class="c5 lst-kix_bvif5itv0aur-0 start">
            <li class="c2 li-bullet-0">
                <span
                    >Suppose we know the optimal costs for all the edges in
                    column </span
                ><span class="c10">i</span>
            </li>
            <li class="c2 li-bullet-0">
                <span>Our task is to find the optimal cost at column </span
                ><span class="c10">i </span><span class="c1">+1</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >If we consider the sites in the lattice then the optimal
                    cost will be:</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 602px;
                        height: 46.67px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image52.png"
                        style="
                            width: 602px;
                            height: 46.67px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >(minimum of either the total cost going up, or total cost
                    going down??)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >This is the defining equation in dynamic programming</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >We have to treat boundary sites specially, but this is just
                    book-keeping</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.50um6w6sysll"><span class="c0">Example</span></h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 450px;
                    height: 442px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image33.png"
                    style="
                        width: 450px;
                        height: 442px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.9r9c6hut61bg">
            <span class="c0">Backward Algorithm</span>
        </h2>
        <ul class="c5 lst-kix_q3utfkf8i288-0 start">
            <li class="c2 li-bullet-0">
                <span
                    >Having found optimal costs c(i,j) we can find the optimal
                    path starting from (</span
                ><span class="c10">n</span><span class="c1">,0)</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >At each step we have a choice of going up or down</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >We choose the direction which satisfies the
                    constraint</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 376px;
                        height: 38px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image46.png"
                        style="
                            width: 376px;
                            height: 38px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >If both direction satisfy the constraint we have more than
                    one optimal path</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.ey7tk1nvbbsh">
            <span class="c0">Time Complexity</span>
        </h2>
        <ul class="c5 lst-kix_cyjxqh2auvho-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >In our solution we had to compute the cost c(i,j) at each
                    lattice point</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span>There were (</span><span class="c10">n</span
                ><span class="c1">/2 + 1)&sup2; lattice point</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >It took constant time to compute each cost so the total
                    time to perform the forward algorithm was
                    &Theta;(n&sup2;)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >The time complexity of the backward algorithm was
                    &Theta;(n)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >This compares with exp(&Theta;(n)) for the brute force
                    algorithm</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.yr8naksd1nm4">
            <span class="c0">Global Matching</span>
        </h2>
        <ul class="c5 lst-kix_gnl6x8gmvymi-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Needleman-Wunsch problem and solution for global
                    matching</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span>Denote costs of a substitution, </span
                ><span class="c10">a</span><span>&rarr;</span
                ><span class="c10">b</span><span>, by s(</span
                ><span class="c10">a</span><span>,</span
                ><span class="c10">b</span><span class="c1">)</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >For proteins we can take this to be our Blosum50
                    matrix</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >There are different substitution matrices e.g. Blosum60
                    used for closely related species</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >We will assume the cost of an insertion or deletion is
                    -8</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.lgw3diwz5wcp">
            <span class="c0">Problem Definition</span>
        </h2>
        <ul class="c5 lst-kix_sgemc2uhmeub-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">We are given two strings:</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_sgemc2uhmeub-1 start">
            <li class="c3 c11 li-bullet-0">
                <span>a = a</span><span class="c8">1</span><span>a</span
                ><span class="c8">2</span><span>&nbsp;&hellip; a</span
                ><span class="c8">m</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span>b = b</span><span class="c8">1</span><span>b</span
                ><span class="c8">2</span><span>&nbsp;&hellip; b</span
                ><span class="c8">n</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_sgemc2uhmeub-0">
            <li class="c2 li-bullet-0">
                <span
                    >We want to find the minimum edit distance measuring how to
                    change </span
                ><span class="c10">a </span><span>to </span
                ><span class="c10">b</span
                ><span>&nbsp;where the price of a substitution is s(</span
                ><span class="c10">a</span><span>,</span
                ><span class="c10">b</span
                ><span class="c1"
                    >) and of an insertion or deletion d = -8</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >To compute this we compute a tableau showing the best
                    possible moves using the dynamic programming formula:</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 476px;
                        height: 127px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image34.png"
                        style="
                            width: 476px;
                            height: 127px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >(value at i,j is the max of either substitution (diagonal),
                    left or right substitution or deletion)</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.kajb1owm8bbw">
            <span class="c0">Example Table</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 456px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image47.png"
                    style="
                        width: 602px;
                        height: 456px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <p class="c3">
            <span class="c1"
                >Then backwards algorithm (basically follow the arrows from the
                bottom right):</span
            >
        </p>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 449.33px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image15.png"
                    style="
                        width: 602px;
                        height: 449.33px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 260px;
                    height: 174px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image44.png"
                    style="
                        width: 260px;
                        height: 174px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.kbtqnn7ek9rv">
            <span class="c0">Space Complexity</span>
        </h2>
        <ul class="c5 lst-kix_5ovu731frlwi-0 start">
            <li class="c2 li-bullet-0">
                <span>For two strings of length </span><span class="c10">m</span
                ><span>&nbsp;and </span><span class="c10">n</span
                ><span
                    >&nbsp;the time complexity of dynamic programming is the
                    size of the tableau O(</span
                ><span class="c10">m n</span><span class="c1">)</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >This is massively faster than exhaustive search</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Unfortunately in biology we often want to match two very
                    long strings (particularly when looking at DNA)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >The space complexity is then more of an issue than time
                    complexity (we don&rsquo;t have enough memory to store the
                    tableau)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Algorithms exist that use linear space complexity</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Can still be too slow for DNA</span>
            </li>
        </ul>
        <h1 class="c7" id="h.xf7s56qyd76h">
            <span class="c12">6 - Local Sequence Matching</span>
        </h1>
        <h2 class="c6" id="h.faew5f34xhlw">
            <span class="c0">Local Alignment</span>
        </h2>
        <ul class="c5 lst-kix_zgonnor7oc6-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">We see global alignment</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >But nature of evolution is that often section from one part
                    of the genome are accidentally copied to another</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Thus, we are often more interested in finding best matching
                    sub-sequences</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Problem known as the Smith-Waterman problem, they first
                    posed and gave an algorithm to solve</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.uqad9tj571y7">
            <span class="c0">Dynamic Programming Solution</span>
        </h2>
        <ul class="c5 lst-kix_k6frrewedr0o-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Surprisingly it&#39;s just a small tweak to global
                    assignment problem</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Add the option of ignoring the match with cost 0</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1">Gives the formula:</span>
            </li>
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 596px;
                        height: 215px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image50.png"
                        style="
                            width: 596px;
                            height: 215px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >(essentially means there are no negative numbers in the
                    table)</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.8s841lxvrkd0">
            <span class="c0">Example Table</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 450.67px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image14.png"
                    style="
                        width: 602px;
                        height: 450.67px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <p class="c3">
            <span class="c1"
                >Then backward algorithm starting from the highest number in the
                grid till zero</span
            >
        </p>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 449.33px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image37.png"
                    style="
                        width: 602px;
                        height: 449.33px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 262px;
                    height: 232px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image20.png"
                    style="
                        width: 262px;
                        height: 232px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.wbeho96fve41">
            <span class="c0">Difference to Global Assignment</span>
        </h2>
        <ul class="c5 lst-kix_32cxpbkv3so-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >The forward algorithm is same except we insist F(i,j) &ge;
                    0</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >We also store the location of the highest F(i,j)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >In the backward algorithm we start at the best value and go
                    back until we reach 0</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Easy to change global to local assignment</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.crmm88n3g2vf">
            <span class="c0">Repeated and Overlapping Matches</span>
        </h2>
        <ul class="c5 lst-kix_bmrnuya7vfgu-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">So far considered single best match</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >In same cases (where have repeated domain or motif) we may
                    have one region being repeated many times within a second
                    sequence</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >We can modify our dynamic programming to allow this</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Again we may believe that one sequence should be completely
                    contained in a second, then we can modify our boundary
                    conditions to find this</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.ojrwsk2k27qk"><span class="c0">Gaps</span></h2>
        <ul class="c5 lst-kix_9r19sgmwwrug-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Insertions are usually caused by adding a subsequence from
                    elsewhere</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Deletions would result from jumping a number of bases or
                    treating an exon as an intron</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Although these are rare events, once it happens then the
                    gap could be quite large</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Up to now we have made a cost of a gap proportional to its
                    length, but this seems unrealistic</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.n4xku8hmc6vw">
            <span class="c0">Dumb Gap Algorithm</span>
        </h2>
        <ul class="c5 lst-kix_vxsk8b23y7c0-0 start">
            <li class="c2 li-bullet-0">
                <span>If we assign a cost &gamma;(</span
                ><span class="c10">g</span><span>) to a gap size </span
                ><span class="c10">g</span
                ><span class="c1"
                    >&nbsp;then we could use the dynamic programme</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 602px;
                        height: 101.33px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image36.png"
                        style="
                            width: 602px;
                            height: 101.33px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
            <li class="c2 li-bullet-0">
                <span>Alas for two sequences of length </span
                ><span class="c10">n</span
                ><span class="c1"
                    >&nbsp;this is a O(n&sup3;) algorithm which is too slow for
                    most practical applications</span
                >
            </li>
            <li class="c2 li-bullet-0"><span class="c1">Better?</span></li>
        </ul>
        <h2 class="c6" id="h.y7bfyldpl3v3">
            <span class="c0">Affine Gap</span>
        </h2>
        <ul class="c5 lst-kix_8zzl2t1m1jcl-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1">If we choose a gap cost of</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_8zzl2t1m1jcl-1 start">
            <li class="c3 c11 li-bullet-0">
                <span>&gamma;(g) = </span><span class="c10">d </span
                ><span>+ (</span><span class="c10">g</span
                ><span>&nbsp;- 1)</span><span class="c10">e</span>
            </li>
        </ul>
        <ul class="c5 lst-kix_8zzl2t1m1jcl-0">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Then there exists a O(n&sup2;) algorithm to do
                    matching</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span>An affine transform of a variable </span
                ><span class="c10">x</span><span>&nbsp;is of the form </span
                ><span class="c10">a x + b </span
                ><span class="c1"
                    >(i.e. a straight line) - note that &gamma;(g) is an affine
                    function</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span>Using </span><span class="c10">d </span
                ><span>= -12 and </span><span class="c10">e</span
                ><span class="c1">&nbsp;= -2 gives a good model</span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >It penalises any gap, with larger gaps being slightly more
                    punished</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.pf952buikcw">
            <span class="c0">Affine Gap Algorithm</span>
        </h2>
        <ul class="c5 lst-kix_fjwuffgkkyk4-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >To efficiently cope with gaps we need to introduce three
                    matrices (rather than just F(i,j))</span
                >
            </li>
        </ul>
        <ul class="c5 lst-kix_fjwuffgkkyk4-1 start">
            <li class="c3 c11 li-bullet-0">
                <span>M(i,j) is the best score up to (i,j) given a</span
                ><span class="c8">i</span><span>&nbsp;and </span
                ><span class="c10">b</span><span class="c8 c10">j</span
                ><span class="c1">&nbsp;align</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span>I</span><span class="c8">a</span
                ><span>(i,j) is the best score given that </span
                ><span class="c10">a</span><span class="c8 c10">i</span
                ><span class="c1">&nbsp;aligns with a gap</span>
            </li>
            <li class="c3 c11 li-bullet-0">
                <span>I</span><span class="c8">b</span
                ><span>(i,j) is the best score given that </span
                ><span class="c10">b</span><span class="c8 c10">j</span
                ><span class="c1">&nbsp;aligns with a gap</span>
            </li>
        </ul>
        <h2 class="c6" id="h.op4a99i5rwau">
            <span class="c0">Update Equation</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 282.67px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image1.png"
                    style="
                        width: 602px;
                        height: 282.67px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <p class="c3 c4"><span class="c1"></span></p>
        <h2 class="c6" id="h.dlgxh7upoyel">
            <span class="c0">Matching with Gaps</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 394.67px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image28.gif"
                    style="
                        width: 602px;
                        height: 394.67px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.rdgwpg96u5of">
            <span class="c0">Different FSA</span>
        </h2>
        <ul class="c5 lst-kix_t4c44m70c5xm-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >The finite automaton (FSA) description provides an elegant
                    visualisation of dynamic programming rules</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >It makes it much easier to come up with different models
                    that perhaps can better capture the evolutionary
                    process</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >But comes with cost that for each new state you need to
                    generate an extra matrix - slowing down matching</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >For affine gaps we can come up with an approximate two
                    state model that nearly always finds the same solution as
                    the three state model</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.asyvqu9u7k08">
            <span class="c0">Inexact Matching in Practice</span>
        </h2>
        <ul class="c5 lst-kix_qiepu7jnuvv6-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Proteins area few 100 amino acids long as O(mn) is
                    doable</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >There are 100s of millions of proteins in the main protein
                    database so it can take a while to look for matches against
                    all known proteins</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >DNA sequences however, are of the order 10&#8309; up to
                    10&sup1;&sup1;</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Using dynamic programming on a large genome is not
                    feasible</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.t0s8rl6z0iuo">
            <span class="c0">Approximate Matching</span>
        </h2>
        <ul class="c5 lst-kix_byy0j030dmzo-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Various techniques have been developed to find approximate
                    matches</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Not guaranteed to find the best inexact match</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Use heuristic of searching for sections with some exact
                    matches</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Once these sections are identified they are extended and
                    combined to find good matches that are polished using
                    Smith-Waterman</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >These algorithms are a rag-bag of heuristic techniques that
                    are very effective if not very elegant</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.2mbtqyqhodl8"><span class="c0">FASTA</span></h2>
        <ul class="c5 lst-kix_st8qq8yef3po-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >FASTA is an approximate matching algorithm for both
                    proteins and nucleotide sequences</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >It firstly checks for short (k-mer) exact matches between a
                    query sequence and a target (k=2 for proteins and k=4-6 for
                    DNA)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Local regions identified with high density of k-mers</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Local regions extended by looking for inexact matches</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Ungapped regions are combined into gapped region</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Finally plausible regions are rematched using dynamical
                    programming</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.4mw1vxxgi3wr"><span class="c0">BLAST</span></h2>
        <ul class="c5 lst-kix_bojofdu03jtr-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >BLAST (Basic Local Alignment Search Tool) is the most
                    commonly used tool for alignment (mainly superseding
                    FASTA)</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >After removing low-complexity regions that might give high
                    matches, exact matches (slightly larger than FASTA) are
                    found</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >BLAST builds up high scoring &ldquo;words&rdquo; from the
                    exact matches</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >These seeds are extended into regions which can then be
                    combined (in BLAST2) to find gapped regions</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Again these are polished with Smith-Waterman</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.jy9ea1ctl0mh">
            <span class="c0">Are Matches Real?</span>
        </h2>
        <ul class="c5 lst-kix_b2y5wndh9q9q-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >By chance we are likely to find some good matches even
                    between random sequences</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >What we would like to know is the probability of finding a
                    particular match by chance</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >This is used in algorithms like BLAST to decide which
                    matched regions it should keep</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >The problem is non-trivial because we are choosing the best
                    match that we found</span
                >
            </li>
        </ul>
        <h2 class="c6" id="h.5sgnsr1hdb74">
            <span class="c0">Probability of a Match</span>
        </h2>
        <ul class="c5 lst-kix_v0l305gsllva-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Computing probability of an exact match between two random
                    strings is possible if we know the frequency of occurrence
                    of all the elements</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >For inexact matching this requires a bit more work</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >The BLOSUM50 scores are designed to give the
                    (log-)probability of a particular match between amino acid
                    residues</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >A lot of effort went into creating matching schemes that
                    give probability of a match</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span>But what is the probability of the best of </span
                ><span class="c10">n</span
                ><span>&nbsp;matches being greater than </span
                ><span class="c10">Z</span><span class="c1">?</span>
            </li>
        </ul>
        <h2 class="c6" id="h.xrmi2okolp77">
            <span class="c0">Extreme Values</span>
        </h2>
        <p class="c3">
            <span
                style="
                    overflow: hidden;
                    display: inline-block;
                    margin: 0px 0px;
                    border: 0px solid #000000;
                    transform: rotate(0rad) translateZ(0px);
                    -webkit-transform: rotate(0rad) translateZ(0px);
                    width: 602px;
                    height: 381.33px;
                "
                ><img
                    alt=""
                    src="assets/computational-biology/image17.gif"
                    style="
                        width: 602px;
                        height: 381.33px;
                        margin-left: 0px;
                        margin-top: 0px;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                    "
                    title=""
            /></span>
        </p>
        <h2 class="c6" id="h.yb4zflt2hqgl">
            <span class="c0">Extreme Value Distribution</span>
        </h2>
        <ul class="c5 lst-kix_by4yyw4cm9w9-0 start">
            <li class="c2 li-bullet-0">
                <span>The distribution of extreme values (best of </span
                ><span class="c10">k</span
                ><span class="c1"
                    >) is known as the extreme value distribution</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Similar to the central limit theorem (which states that the
                    sum of random variables will be normally distributed), it
                    turns out that asymptotically (k&rarr;&infin;) the extreme
                    value of random deviates converges to a small number of
                    extreme value distributions</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >The most common is the Gumbel EVD given by:</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 406px;
                        height: 136px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image10.png"
                        style="
                            width: 406px;
                            height: 136px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
        </ul>
        <h2 class="c6" id="h.4j4rjggzwj9o">
            <span class="c0">Using EVD</span>
        </h2>
        <ul class="c5 lst-kix_8vkv9emkr37f-0 start">
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >Extreme value distributions become more accurate when we do
                    a large number of trials</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 557.5px;
                        height: 233.37px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image53.png"
                        style="
                            width: 557.5px;
                            height: 233.37px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
            <li class="c2 li-bullet-0">
                <span class="c1"
                    >In matching the number of matches are exponential in the
                    string length so we are in a situation where EVDs are quite
                    accurate</span
                >
            </li>
            <li class="c2 li-bullet-0">
                <span
                    >The probability of observing a match with a score at least </span
                ><span class="c10">Z</span><span class="c1">&nbsp;is:</span>
            </li>
            <li class="c2 li-bullet-0">
                <span
                    style="
                        overflow: hidden;
                        display: inline-block;
                        margin: 0px 0px;
                        border: 0px solid #000000;
                        transform: rotate(0rad) translateZ(0px);
                        -webkit-transform: rotate(0rad) translateZ(0px);
                        width: 602px;
                        height: 89.33px;
                    "
                    ><img
                        alt=""
                        src="assets/computational-biology/image27.png"
                        style="
                            width: 602px;
                            height: 89.33px;
                            margin-left: 0px;
                            margin-top: 0px;
                            transform: rotate(0rad) translateZ(0px);
                            -webkit-transform: rotate(0rad) translateZ(0px);
                        "
                        title=""
                /></span>
            </li>
        </ul>
    </body>
</html>
